docs/proposals/fp2.rst

```diff
@@ -0,0 +1,145 @@
+FP2: Local Lint Rules
+=====================
+
+By default, Fixit expects rules to be referenced by their fully-qualified
+module and/or class name. For example, ``fixit.rules`` refers to the bundled set
+if lint rules shipped with Fixit. Third party rules available in the environment
+can similarly be referenced by their module name, as long as they are installed
+and importable by Fixit at runtime. These will be collectively referred to as
+"global rules" here for sake of clarity.
+
+However, in many cases, it is also useful for projects using Fixit to build
+custom lint rules specific to the project being linted. It is beneficial for
+those rules to be defined within the codebase being linted, without needing to
+first install those lint rules into the environment before running Fixit.
+
+These are defined as "local rules", and will require dedicated behavior from
+Fixit to discover and import these rules at runtime, as well as special
+configuration syntax for enabling these local rules in a project.
+
+
+Configuration
+-------------
+
+When configuring the set of enabled or disabled rules, any local rules must
+be marked with a single leading period, and referred to using their path
+relative to the configuration file's directory.
+
+For example, a configuration file at ``project/fixit.toml`` could include
+rules defined in ``project/some/local/rules.py`` with the following:
+
+.. code-block:: toml
+
+    [tool.fixit]
+    enable = [".some.local.rules"]
+
+
+References to local rules should be accepted in either ``enable`` or ``disable``
+options, including overrides, to provide the same selection criteria available
+to global rules:
+
+.. code-block:: toml
+
+    [tool.fixit]
+    enable = [".rules"]
+
+    [[tool.fixit.overrides]]
+    path = "project1"
+    disable = [".rules.PickyRule"]
```

Comment:
AIUI inheritable configs mean this as well as the parent might be able to `enable = [".rules"]` -- if that's correct, which one would this disable?

Reply:
`".rules"` would refer to the same thing within a single file (ie, local rules in overrides are not affected by the subpath). So the top-level `enable = [".rules"]` and the override's `enable = [".rules"]` are enabling the exact same local rules module, and therefore the `disable = [".rules.PickyRule"]` is similarly deterministic.

Reply:
I don't know that we're talking about the same thing.  Let's say we have

```
fixit.toml
rules/GoodRule.py
rules/PickyRule.py
subdir/fixit.toml
subdir/rules/ThirdRule.py
```

Is there a way for toplevel fixit.toml to enable `.rules` and subdir to disable toplevel's `.rules.PickyRule`?  I am interpreting this text from FP1 when assuming that toplevel's enables would be inherited, with emphasis added: "or by _separate config files within those subpaths_. Overrides should be applied to all files within the relative subpaths, accounting for further nested overrides."

Reply:
No, the proposal is that a config in a subdirectory would not be able to disable local rules defined in a parent directory's config; the suggested method is either to use overrides in that parent config, or the escape hatch of setting `root=true` and abandoning the parent configs entirely.

This is mainly because I couldn't imagine a reasonable way to define exactly what level of local rules are being enabled/disabled without getting into weird syntax or structure that ended up being more complicated to document and implement. If you have suggestions for how to do this without ambiguity, please share. :)